src/lifecycles/RoomViewLifecycle.ts

```diff
@@ -28,7 +28,7 @@ export type RoomPreviewOpts = {
 
 export type ViewRoomOpts = {
     buttons: Array<{
-        icon: React.ReactNode;
+        icon: () => React.ReactNode;
```

Comment:
This is a breaking change, please offer a backwards compatible approach instead.